pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -795,6 +795,15 @@ allChainsLabel:
 			filterChainMatch = &fcm
 			if filterChainMatchOption[id].Protocol == istionetworking.ListenerProtocolHTTP {
 				httpOpts = configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, pluginParams)
+				if chain.TLSContext != nil && chain.TLSContext.CommonTlsContext != nil {
```

Comment:
refactor to make it clear, add function
`dropAlpnFromList(list, alpn_to_drop)`

Reply:
Thanks..Fixed